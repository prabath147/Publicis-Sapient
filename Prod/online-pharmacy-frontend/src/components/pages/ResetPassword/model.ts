export interface ResetPasswordForm {
    oldPassword: string;
    newPassword: string;
    confirmPassword: string;
}

export interface ResetPasswordRequest {
    oldPassword: string;
    newPassword: string;
  }