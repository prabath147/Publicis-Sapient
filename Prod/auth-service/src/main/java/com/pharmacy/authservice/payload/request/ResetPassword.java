package com.pharmacy.authservice.payload.request;

import lombok.*;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.Pattern;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Builder
public class ResetPassword {
    @NotBlank(message = "old password may not be empty or null")
    String oldPassword;
    @NotBlank(message = "new password may not be empty or null")
    @Pattern(regexp = "^(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z])(?=.*[@#$%!^&+=])(?=\\S+$).{8,}$", message = "Must contain at least one number and one uppercase and lowercase letter and one special character, and at least 8 or more characters")
    String newPassword;

}
