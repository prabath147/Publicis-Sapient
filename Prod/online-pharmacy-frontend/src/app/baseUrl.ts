export const baseURL = "https://frk5iphcfk.execute-api.us-east-1.amazonaws.com/";
    // "http://localhost:8081";
    
