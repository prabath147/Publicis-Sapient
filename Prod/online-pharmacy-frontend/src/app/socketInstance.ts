
export const socketURL = "https://c8zpvs7fv4.execute-api.us-east-1.amazonaws.com/notify-ws"

// const socket = new SockJS(socketURL);
// export const stompClient = Stomp.over(socket);