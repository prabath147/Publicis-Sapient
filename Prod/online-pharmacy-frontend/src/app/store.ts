import { Action, configureStore, ThunkAction } from "@reduxjs/toolkit";
import StoreReducer from "../components/pages/Admin/storeList/StoreSlice";
import SubscriptionReducer from "../components/pages/AdminSubscriptionPack/SubscriptionvSlice";
import UserReducer from "../components/pages/Auth/Login/UserSlice";
import CartReducer from "../components/pages/Cart/CartSlice";
import ManagerReducer from "../components/pages/Manager/ManagerSlice";
import NotificationReducer from "../components/pages/Open/notificationList/NotificationSlice";
// import OrderSlice from "../components/pages/orderDetail/OrderSlice";
// import OrderListSlice from "../components/pages/orderList/OrderListSlice";

export const store = configureStore({
  reducer: {
    storeL: StoreReducer,
    user: UserReducer,
    notification: NotificationReducer,
    viewsubscription: SubscriptionReducer,
    // addsubscription: AddSubcriptionRe,
    cart: CartReducer,
    // order: OrderSlice,
    // orderL: OrderListSlice,
    manager: ManagerReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
