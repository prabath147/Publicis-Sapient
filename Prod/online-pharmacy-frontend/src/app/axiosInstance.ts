import axios, { AxiosRequestConfig } from "axios";
import LocalUserService from "../services/LocalUserService";
import TokenService from "../services/TokenService";
import { baseURL } from "./baseUrl";

const axiosInstance = axios.create({
  baseURL: baseURL,
  timeout: 5000,
  headers: {
    "Content-Type": "application/json",
    accept: "application/json",
  },
});

axiosInstance.interceptors.request.use(
  (config: AxiosRequestConfig) => {
    const token = TokenService.getLocalAccessToken();
    if (token && config.headers !== undefined) {
      config.headers["Authorization"] = "Bearer " + token; // for Spring Boot back-end
      // config.headers["x-access-token"] = token; // for Node.js Express back-end
      config.headers["ngrok-skip-browser-warning"] = "";
    }
    return config;
  },
  (error) => {
    return Promise.reject(error);
  }
);

axiosInstance.interceptors.response.use(
  (res) => {
    return res;
  },
  async (err) => {
    const originalConfig = err.config;
    if (originalConfig.url !== "api/auth/signin" && err.response) {
      // Access Token was expired
      if (err.response.status === 401 && !originalConfig._retry) {
        originalConfig._retry = true;
        try {
          const rs = await axiosInstance.post("api/auth/refresh-token", {
            refreshToken: TokenService.getLocalRefreshToken(),
          });
          const { accessToken } = rs.data;
          TokenService.updateLocalAccessToken(accessToken);
          return axiosInstance(originalConfig);
        } catch (_error) {
          TokenService.removeLocalAccessToken()
          TokenService.removeLocalRefreshToken()
          LocalUserService.removeLocalUser()
          // window.location.reload();
          return Promise.reject(_error);
        }
      }
    }
    return Promise.reject(err);
  }
);
export default axiosInstance;
