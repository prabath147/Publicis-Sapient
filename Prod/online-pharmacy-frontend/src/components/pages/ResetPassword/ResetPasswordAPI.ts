import axiosInstance from "./../../../app/axiosInstance";
// import { useAppSelector } from '../../../app/hooks';
// import { getUserData } from "../Auth/Login/UserSlice";
import { ResetPasswordRequest } from "./model";

export const resetPasswordAPI = (userId: number, data:ResetPasswordRequest) =>
  axiosInstance.post(`api/auth/reset-password/${userId}`, data);

