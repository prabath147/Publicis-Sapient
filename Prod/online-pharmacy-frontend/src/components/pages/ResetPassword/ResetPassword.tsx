import { useForm } from "@mantine/form";
import { useNavigate } from "react-router-dom";
import { useAppSelector } from "../../../app/hooks";
import { ResetPasswordForm, ResetPasswordRequest } from "./model";

import { Button, Card, Container, PasswordInput, Title } from "@mantine/core";

import { showNotification } from "@mantine/notifications";
import { IconLock } from "@tabler/icons";
import { useDispatch } from "react-redux";
import AuthNavbar from "../../ui/AuthNavbar/AuthNavbar";
import { logOutAPI } from "../../ui/UserBadge/UserBadgeAPI";
import { getUserData, LogoutAction } from "../Auth/Login/UserSlice";
import { checkPassword } from "../Auth/Registration/RegistrationUtil";
import { resetPasswordAPI } from "./ResetPasswordAPI";

export default function ResetPassword() {
  const navigate = useNavigate();
  const userId = useAppSelector(getUserData).id;
  const dispatch = useDispatch();

  const form = useForm({
    initialValues: { oldPassword: "", newPassword: "", confirmPassword: "" },
    validate: {
      newPassword: (value) =>
        checkPassword(value)
          ? null
          : "Must contain at least one number and one uppercase and lowercase letter, and at least 8 or more characters",
      confirmPassword: (value) =>
        checkPassword(value)
          ? null
          : "Must contain at least one number and one uppercase and lowercase letter, and at least 8 or more characters",
    },
  });

  async function resetPassword(data: ResetPasswordForm) {
    if (data.newPassword !== data.confirmPassword) {
      showNotification({
        message: "password confirmation failed, please provide same password",
        color: "red",
      });
      return;
    }

    const passwordRequest: ResetPasswordRequest = {
      oldPassword: data.oldPassword,
      newPassword: data.newPassword,
    };

    resetPasswordAPI(userId, passwordRequest)
      .then((response) => {
        showNotification({
          message: "Password Reset Successfully",
          color: "green",
        });
        logOutAPI()
          .then(() => {
            dispatch(LogoutAction());
          })
          .catch(() => {
            dispatch(LogoutAction());
          });
        navigate("/login");
      })
      .catch((error) => {
        console.log(error);
        showNotification({ message: error.response.data, color: "red" });
      });
  }

  return (
    <>
      <AuthNavbar />
      <br />
      <br />
      <Title order={2} align="center" color="blue.6">
        Reset your Password!!
      </Title>
      <Container size={420} my={40}>
        <Card shadow="sm" p="lg" radius="md" withBorder>
          <form onSubmit={form.onSubmit(resetPassword)}>
            <PasswordInput
              icon={<IconLock size={16} />}
              label="Old Password"
              placeholder="Old password"
              {...form.getInputProps("oldPassword")}
            />
            <PasswordInput
              icon={<IconLock size={16} />}
              mt="sm"
              label="New password"
              placeholder="New password"
              {...form.getInputProps("newPassword")}
            />
            <PasswordInput
              icon={<IconLock size={16} />}
              mt="sm"
              label="Confirm password"
              placeholder="Confirm password"
              {...form.getInputProps("confirmPassword")}
            />

            <Button type="submit" mt="sm" fullWidth>
              Submit
            </Button>
          </form>
        </Card>
      </Container>
    </>
  );
}
