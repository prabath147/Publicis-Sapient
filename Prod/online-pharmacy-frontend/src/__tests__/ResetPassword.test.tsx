import { cleanup, render, screen } from "@testing-library/react";
import user from "@testing-library/user-event";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { store } from "../app/store";
import ResetPassword from "../components/pages/ResetPassword/ResetPassword";
import mockAxios from "../__mocks__/axios";

const MockResetPwd = () => {
  return (
    <Provider store={store}>
      <BrowserRouter>
        <ResetPassword />
      </BrowserRouter>
    </Provider>
  );
};

it("component created", () => {
  const component = renderer.create(<MockResetPwd />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

describe("Reset Password Tests", () => {
  afterEach(() => {
    cleanup();
  });
  it("Check if form is rendered correctly", () => {
    render(<MockResetPwd />);

    // expect(
    //   screen.getByRole("textbox", { name: /User Name/i })
    // ).toBeInTheDocument();
    // expect(screen.getByLabelText(/password/i)).toBeInTheDocument();
    // expect(
    //   screen.getByRole("button", { name: /Sign in/i })
    // ).toBeInTheDocument();

    expect(screen.getByLabelText(/Old Password/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/New password/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/Confirm password/i)).toBeInTheDocument();
  });

  it("should fill form1", async () => {
    render(<MockResetPwd />);

    const oldpwdInput = screen.getByLabelText(/Old Password/i);
    const newpwdInput = screen.getByLabelText(/New password/i);
    const confirmpassWordInput = screen.getByLabelText(/Confirm password/i);

    await user.type(oldpwdInput, "Test@123");
    expect(oldpwdInput).toHaveValue("Test@123");

    await user.type(newpwdInput, "NewTest@123");
    expect(newpwdInput).toHaveValue("NewTest@123");

    await user.type(confirmpassWordInput, "NewTest@123");
    expect(confirmpassWordInput).toHaveValue("NewTest@123");
  }, 30000);

  it("should fill incorrect", async () => {
    render(<MockResetPwd />);

    const oldpwdInput = screen.getByLabelText(/Old Password/i);
    const newpwdInput = screen.getByLabelText(/New password/i);
    const confirmpassWordInput = screen.getByLabelText(/Confirm password/i);

    const subBtn = screen.getByRole("button", {
      name: /submit/i,
    });

    await user.type(oldpwdInput, "Test@123");

    await user.type(newpwdInput, "newtest123");
    await user.type(confirmpassWordInput, "newtest123");

    // mockAxios.post.mockImplementationOnce(() => Promise.resolve());

    await user.click(subBtn);
  }, 30000);

  it("should fill different confirm", async () => {
    render(<MockResetPwd />);

    const oldpwdInput = screen.getByLabelText(/Old Password/i);
    const newpwdInput = screen.getByLabelText(/New password/i);
    const confirmpassWordInput = screen.getByLabelText(/Confirm password/i);

    const subBtn = screen.getByRole("button", {
      name: /submit/i,
    });

    await user.type(oldpwdInput, "Test@123");

    await user.type(newpwdInput, "NewTest@123");
    await user.type(confirmpassWordInput, "Newtest@123");

    const response = {
      data: "USUADUAS",
    };

    mockAxios.post.mockImplementationOnce(() => Promise.resolve());

    await user.click(subBtn);
  }, 30000);

  it("should fill correct", async () => {
    render(<MockResetPwd />);

    const oldpwdInput = screen.getByLabelText(/Old Password/i);
    const newpwdInput = screen.getByLabelText(/New password/i);
    const confirmpassWordInput = screen.getByLabelText(/Confirm password/i);

    const subBtn = screen.getByRole("button", {
      name: /submit/i,
    });

    await user.type(oldpwdInput, "Test@123");

    await user.type(newpwdInput, "NewTest@123");
    await user.type(confirmpassWordInput, "NewTest@123");

    const response = {
      data: "USUADUAS",
    };

    mockAxios.post.mockImplementationOnce(() => Promise.resolve());

    await user.click(subBtn);
  }, 30000);

  it("should fill correct with reject ", async () => {
    render(<MockResetPwd />);

    const oldpwdInput = screen.getByLabelText(/Old Password/i);
    const newpwdInput = screen.getByLabelText(/New password/i);
    const confirmpassWordInput = screen.getByLabelText(/Confirm password/i);

    const subBtn = screen.getByRole("button", {
      name: /submit/i,
    });

    await user.type(oldpwdInput, "Test@123");

    await user.type(newpwdInput, "NewTest@123");
    await user.type(confirmpassWordInput, "NewTest@123");

    const response = {
      response: { data: "USUADUAS" },
    };

    mockAxios.post.mockImplementationOnce(() => Promise.reject(response));

    await user.click(subBtn);
  }, 30000);
});
