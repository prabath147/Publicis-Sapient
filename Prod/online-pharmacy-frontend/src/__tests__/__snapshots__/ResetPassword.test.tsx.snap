// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component created 1`] = `
Array [
  <div
    className="mantine-Container-root mantine-15pjuqq"
  >
    <div
      className="main-nav"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <div
          className="mantine-Image-root mantine-15po0m8"
          style={
            Object {
              "height": "auto",
              "width": "100%",
            }
          }
        >
          <figure
            className="mantine-11nhzn5 mantine-Image-figure"
          >
            <div
              className="mantine-qqmv3w mantine-Image-imageWrapper"
            >
              <img
                className="mantine-3y8yz3 mantine-Image-image"
                onError={[Function]}
                onLoad={[Function]}
                src="logo.jpg"
                style={
                  Object {
                    "height": "auto",
                    "objectFit": "cover",
                    "width": "100%",
                  }
                }
              />
            </div>
          </figure>
        </div>
      </a>
    </div>
  </div>,
  <br />,
  <br />,
  <h2
    className="mantine-Text-root mantine-Title-root mantine-1yvlqai"
  >
    Reset your Password!!
  </h2>,
  <div
    className="mantine-Container-root mantine-1tg98w8"
  >
    <div
      className="mantine-Paper-root mantine-Card-root mantine-ergsfy"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="mantine-InputWrapper-root mantine-PasswordInput-root mantine-1m3pqry"
        >
          <label
            className="mantine-InputWrapper-label mantine-PasswordInput-label mantine-ittua2"
            htmlFor="mantine-r0"
            id="mantine-r0-label"
          >
            Old Password
          </label>
          <div
            className="mantine-Input-wrapper mantine-PasswordInput-wrapper mantine-12sbrde"
          >
            <div
              className="mantine-yxyjyg mantine-Input-icon mantine-PasswordInput-icon"
            >
              <svg
                className="icon icon-tabler icon-tabler-lock"
                fill="none"
                height={16}
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={2}
                viewBox="0 0 24 24"
                width={16}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                  stroke="none"
                />
                <rect
                  height={10}
                  rx={2}
                  width={14}
                  x={5}
                  y={11}
                />
                <circle
                  cx={12}
                  cy={16}
                  r={1}
                />
                <path
                  d="M8 11v-4a4 4 0 0 1 8 0v4"
                />
              </svg>
            </div>
            <div
              aria-invalid={false}
              className="mantine-PasswordInput-input mantine-Input-input mantine-PasswordInput-input mantine-Input-withIcon mantine-PasswordInput-withIcon mantine-1vsi2ce"
            >
              <input
                className="mantine-PasswordInput-innerInput mantine-PasswordInput-withIcon mantine-ya9vkc"
                id="mantine-r0"
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Old password"
                type="password"
                value=""
              />
            </div>
            <div
              className="mantine-o3oqoy mantine-Input-rightSection mantine-PasswordInput-rightSection"
            >
              <button
                aria-hidden={true}
                className="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-PasswordInput-visibilityToggle mantine-1dcetaa"
                disabled={false}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                <svg
                  fill="none"
                  height={15}
                  viewBox="0 0 15 15"
                  width={15}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clipRule="evenodd"
                    d="M7.5 11C4.80285 11 2.52952 9.62184 1.09622 7.50001C2.52952 5.37816 4.80285 4 7.5 4C10.1971 4 12.4705 5.37816 13.9038 7.50001C12.4705 9.62183 10.1971 11 7.5 11ZM7.5 3C4.30786 3 1.65639 4.70638 0.0760002 7.23501C-0.0253338 7.39715 -0.0253334 7.60288 0.0760014 7.76501C1.65639 10.2936 4.30786 12 7.5 12C10.6921 12 13.3436 10.2936 14.924 7.76501C15.0253 7.60288 15.0253 7.39715 14.924 7.23501C13.3436 4.70638 10.6921 3 7.5 3ZM7.5 9.5C8.60457 9.5 9.5 8.60457 9.5 7.5C9.5 6.39543 8.60457 5.5 7.5 5.5C6.39543 5.5 5.5 6.39543 5.5 7.5C5.5 8.60457 6.39543 9.5 7.5 9.5Z"
                    fill="currentColor"
                    fillRule="evenodd"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <div
          className="mantine-InputWrapper-root mantine-PasswordInput-root mantine-e8sh7b"
        >
          <label
            className="mantine-InputWrapper-label mantine-PasswordInput-label mantine-ittua2"
            htmlFor="mantine-r1"
            id="mantine-r1-label"
          >
            New password
          </label>
          <div
            className="mantine-Input-wrapper mantine-PasswordInput-wrapper mantine-12sbrde"
          >
            <div
              className="mantine-yxyjyg mantine-Input-icon mantine-PasswordInput-icon"
            >
              <svg
                className="icon icon-tabler icon-tabler-lock"
                fill="none"
                height={16}
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={2}
                viewBox="0 0 24 24"
                width={16}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                  stroke="none"
                />
                <rect
                  height={10}
                  rx={2}
                  width={14}
                  x={5}
                  y={11}
                />
                <circle
                  cx={12}
                  cy={16}
                  r={1}
                />
                <path
                  d="M8 11v-4a4 4 0 0 1 8 0v4"
                />
              </svg>
            </div>
            <div
              aria-invalid={false}
              className="mantine-PasswordInput-input mantine-Input-input mantine-PasswordInput-input mantine-Input-withIcon mantine-PasswordInput-withIcon mantine-1vsi2ce"
            >
              <input
                className="mantine-PasswordInput-innerInput mantine-PasswordInput-withIcon mantine-ya9vkc"
                id="mantine-r1"
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="New password"
                type="password"
                value=""
              />
            </div>
            <div
              className="mantine-o3oqoy mantine-Input-rightSection mantine-PasswordInput-rightSection"
            >
              <button
                aria-hidden={true}
                className="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-PasswordInput-visibilityToggle mantine-1dcetaa"
                disabled={false}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                <svg
                  fill="none"
                  height={15}
                  viewBox="0 0 15 15"
                  width={15}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clipRule="evenodd"
                    d="M7.5 11C4.80285 11 2.52952 9.62184 1.09622 7.50001C2.52952 5.37816 4.80285 4 7.5 4C10.1971 4 12.4705 5.37816 13.9038 7.50001C12.4705 9.62183 10.1971 11 7.5 11ZM7.5 3C4.30786 3 1.65639 4.70638 0.0760002 7.23501C-0.0253338 7.39715 -0.0253334 7.60288 0.0760014 7.76501C1.65639 10.2936 4.30786 12 7.5 12C10.6921 12 13.3436 10.2936 14.924 7.76501C15.0253 7.60288 15.0253 7.39715 14.924 7.23501C13.3436 4.70638 10.6921 3 7.5 3ZM7.5 9.5C8.60457 9.5 9.5 8.60457 9.5 7.5C9.5 6.39543 8.60457 5.5 7.5 5.5C6.39543 5.5 5.5 6.39543 5.5 7.5C5.5 8.60457 6.39543 9.5 7.5 9.5Z"
                    fill="currentColor"
                    fillRule="evenodd"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <div
          className="mantine-InputWrapper-root mantine-PasswordInput-root mantine-e8sh7b"
        >
          <label
            className="mantine-InputWrapper-label mantine-PasswordInput-label mantine-ittua2"
            htmlFor="mantine-r2"
            id="mantine-r2-label"
          >
            Confirm password
          </label>
          <div
            className="mantine-Input-wrapper mantine-PasswordInput-wrapper mantine-12sbrde"
          >
            <div
              className="mantine-yxyjyg mantine-Input-icon mantine-PasswordInput-icon"
            >
              <svg
                className="icon icon-tabler icon-tabler-lock"
                fill="none"
                height={16}
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={2}
                viewBox="0 0 24 24"
                width={16}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                  stroke="none"
                />
                <rect
                  height={10}
                  rx={2}
                  width={14}
                  x={5}
                  y={11}
                />
                <circle
                  cx={12}
                  cy={16}
                  r={1}
                />
                <path
                  d="M8 11v-4a4 4 0 0 1 8 0v4"
                />
              </svg>
            </div>
            <div
              aria-invalid={false}
              className="mantine-PasswordInput-input mantine-Input-input mantine-PasswordInput-input mantine-Input-withIcon mantine-PasswordInput-withIcon mantine-1vsi2ce"
            >
              <input
                className="mantine-PasswordInput-innerInput mantine-PasswordInput-withIcon mantine-ya9vkc"
                id="mantine-r2"
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Confirm password"
                type="password"
                value=""
              />
            </div>
            <div
              className="mantine-o3oqoy mantine-Input-rightSection mantine-PasswordInput-rightSection"
            >
              <button
                aria-hidden={true}
                className="mantine-UnstyledButton-root mantine-ActionIcon-root mantine-PasswordInput-visibilityToggle mantine-1dcetaa"
                disabled={false}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                tabIndex={-1}
                type="button"
              >
                <svg
                  fill="none"
                  height={15}
                  viewBox="0 0 15 15"
                  width={15}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clipRule="evenodd"
                    d="M7.5 11C4.80285 11 2.52952 9.62184 1.09622 7.50001C2.52952 5.37816 4.80285 4 7.5 4C10.1971 4 12.4705 5.37816 13.9038 7.50001C12.4705 9.62183 10.1971 11 7.5 11ZM7.5 3C4.30786 3 1.65639 4.70638 0.0760002 7.23501C-0.0253338 7.39715 -0.0253334 7.60288 0.0760014 7.76501C1.65639 10.2936 4.30786 12 7.5 12C10.6921 12 13.3436 10.2936 14.924 7.76501C15.0253 7.60288 15.0253 7.39715 14.924 7.23501C13.3436 4.70638 10.6921 3 7.5 3ZM7.5 9.5C8.60457 9.5 9.5 8.60457 9.5 7.5C9.5 6.39543 8.60457 5.5 7.5 5.5C6.39543 5.5 5.5 6.39543 5.5 7.5C5.5 8.60457 6.39543 9.5 7.5 9.5Z"
                    fill="currentColor"
                    fillRule="evenodd"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <button
          className="mantine-UnstyledButton-root mantine-Button-root mantine-tgyipv"
          data-button={true}
          type="submit"
        >
          <div
            className="mantine-3xbgk5 mantine-Button-inner"
          >
            <span
              className="mantine-qo1k2 mantine-Button-label"
              style={
                Object {
                  "textTransform": undefined,
                }
              }
            >
              Submit
            </span>
          </div>
        </button>
      </form>
    </div>
  </div>,
]
`;
